src/foam/nanos/crunch/ServerCrunchService.java

```diff
@@ -373,7 +370,7 @@ public UserCapabilityJunction getJunctionForSubject(
     AuthService auth = (AuthService) x.get("auth");
     if ( auth.check(x, "service.crunchService.updateUserContext") ) {
       x = Auth.sudo(x, subject.getUser(), subject.getRealUser());
-    }
+    } else throw new AuthorizationException("You don't have permission to check for UCJs");
```

Comment:
It's possible that the call is to this function just as a default -- in case ops is asking for data ... so this exception only makes sense in that case. Ops user trying to access but doesn't have access - they should go no further.
However,
If the call is coming for the client for the client the logic would still call this function but the user doesn't have access to see anyone else's ucj's so doesn't switch context but relies on the default subject.